ddl/ddl_api.go

```diff
@@ -224,18 +224,27 @@ func ResolveCharsetCollation(tblCharset, dbCharset string) (string, string, erro
 	return charset, collate, nil
 }
 
+func typesNeedCharset(tp byte) bool {
```

Comment:
I think this name is a bit ambiguous. Other types also need charset.

Reply:
`typesNeedSetCharset`?

Reply:
IMHO this name is OK, there is `has_charset()` accessor in MySQL's field definition: https://github.com/mysql/mysql-server/blob/ca94b993454c86be248fbe180db94647488114e9/sql/field.h#L1565

But I think it's better to move this function to `parser`(later)

Reply:
Both ok for me. currently is `typesNeedSetCharset`. So which one is better?

Reply:
I don't think the name is good, but I don't have any better suggestions, so I can accept both.